src/services/smartWallet.js

```diff
@@ -79,6 +81,24 @@ export const parseEstimatePayload = (estimatePayload: EstimatePayload): ParsedEs
   };
 };
 
+const calculateEstimate = (
+  estimate,
+  gasInfo?: GasInfo,
+  speed?: string = SPEED_TYPES.NORMAL,
+  defaultGasAmount?: number = DEFAULT_GAS_LIMIT,
+): BigNumber => {
+  let { gasAmount, gasPrice } = parseEstimatePayload(estimate);
+
+  if (!gasAmount) {
+    gasAmount = new BigNumber(defaultGasAmount);
+  }
+  if (!gasPrice) {
+    const defaultGasPrice = get(gasInfo, `gasPrice.${speed}`, 0);
+    gasPrice = utils.parseUnits(defaultGasPrice.toString(), 'gwei');
+  }
+  return gasPrice.mul(gasAmount);
+};
```

Comment:
Can we add some tests for this function considering the scenarios it tries to cover?

Reply:
OK 👍

Reply:
Will add tests later once back next week, feel free to merge if else looks fine.